### ISSUE TITLE
feature_request: code_folding for a specific chunk

### ISSUE THREAD

**Issue by lepennec** [rstudio/rmarkdown#664]
I'm using a lot rmarkdown to construct labs for my student. I've seen the new code_folding option and I was wondering if this option could be specified/modified for some specific chunks. I'm considering using this option so that some hints/solutions are invisible until the students request them.
Thanks,
Erwan

**Comment by diegocgaona** [comment]
Would be nice if we could fold chunks of text too, not only code. That way we can give details or descriptions about some data/code only if the user want more details.

**Comment by mbojan** [comment]
It will be great to have this!!! 

I was playing around with this to be able to fold just the source of a code chunk wen using `github_document` output format. It is possible, if you redefine the source output hook with:

```r
knitr::knit_hooks$set(
  source = function(x, options) {
    hook.r = function(x, options) {
      fence <- "```"
      language = tolower(options$engine)
      if (language == 'node') language = 'javascript'
      if (!options$highlight) language = 'text'
      if(!is.null(options$foldcode)) {
      paste0('\n\n', "<details><summary>Source</summary>\n", fence, language, '\n', x, fence,  '\n\n', "</details>\n")
      } else {
              paste0('\n\n', fence, language, '\n', x, fence,  '\n\n')
      }
    }
    x = knitr:::hilight_source(x, 'markdown', options)
    hook.r(
      paste(c(
        x, 
        ''
      ), collapse = '\n'), 
      options
    )
  }
)
```

and then in `Rmd`, say:

```{r, foldcode=TRUE}
data.frame(x=1:5)
knitr::kable(d)
```

It is a hack because the above hook does not respect the environment created when `render_markdown()` is called -- default hooks depend on objects outside of the hook functions and their arguments, namely: functions `hook.t` and `hook.r`, and objects `strict`, `fence` and `fence_char`. The hook above basically defines/hardcodes these within the hook itself.

**Comment by jeff-goldsmith** [comment]
Not really a solution, but a workaround I've been using to hide hints in labs until students ask for them is the [details](https://www.w3schools.com/tags/tag_details.asp) element in html. Mentioning in case this works for others until a real solution exists.

**Comment by yihui** [comment]
This has been implemented by @atusy via #1602. Thanks!
